src/sync.js

```diff
@@ -460,6 +462,13 @@ SyncApi.prototype.sync = function() {
         // no push rules for guests, no access to POST filter for guests.
         self._sync({});
     } else {
+        // Don't do an HTTP hit to /sync. Instead, load up the persisted /sync data,
```

Comment:
this comment is a bit misleading. It might be better phrased as "before we start fetching the push rules and doing /sync hits, see if we have persisted /sync data and use it if so"